frontend/packages/dev-console/src/components/health-checks/ProbeForm.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikValues, useFormikContext } from 'formik';
+import { TextInputTypes, InputGroupText } from '@patternfly/react-core';
+import { InputGroupField, InputField, DropdownField, ActionGroupWithIcons } from '@console/shared';
+import { HTTPProbeTypeForm, TCPProbeTypeForm, CommandProbeTypeForm } from './ProbeTypeForms';
+import { ProbeType } from './health-checks-types';
+import FormSection from '../import/section/FormSection';
+import './ProbeForm.scss';
+
+const getProbeTypeForm = (value: string, ports: object, probe: string) => {
+  switch (value) {
+    case ProbeType.HTTPGET:
+      return <HTTPProbeTypeForm ports={ports} probe={probe} />;
+    case ProbeType.ContainerCommand:
+      return <CommandProbeTypeForm probe={probe} />;
+    case ProbeType.TCPSocket:
+      return <TCPProbeTypeForm ports={ports} probe={probe} />;
+    default:
+      return undefined;
+  }
+};
+
+interface ProbeFormProps {
+  containerPorts?: object;
+  onSubmit: () => void;
+  onClose: () => void;
+  probe: string;
+}
+
+enum probeTypeOptions {
+  'httpGet' = 'HTTP GET',
+  'containerCommand' = 'Container Command',
+  'tcpSocket' = 'TCP Socket',
+}
```

Comment:
no need to use string for keys
```suggestion
enum ProbeTypeOptions {
  httpGet = 'HTTP GET',
  containerCommand = 'Container Command',
  tcpSocket = 'TCP Socket',
}
```